### ISSUE TITLE
Add Markdown Renderer

### ISSUE THREAD

**Issue by rjwignar** [tarasglek/chatcraft.org#585]
ChatCraft render previews of [Mermaid](https://chatcraft.org/api/share/rjwignar/-AW3hvL0GI29p8hep4z3a) and Nomnoml.
However, I noticed we don't render [Markdown](https://chatcraft.org/c/rjwignar/9Dkbq_a17U4Grso2WQutb):
https://github.com/tarasglek/chatcraft.org/blob/7cc1e2f3ec1aaf77014a56703d7d8e77c706f72d/src/components/Markdown.tsx#L98-L117

Is this a feature we want?

**Comment by mingming-ma** [comment]
I think currently the `ReactMarkdown` from `react-markdown` control the whole render result ( [react-markdown](https://github.com/remarkjs/react-markdown?tab=readme-ov-file#react-markdown) ), it might be very difficult to change the lib but there are tons of plugins to adjust the render results I believe

**Comment by tarasglek** [comment]
yeah it's frustrating that we dont support nested markdown code blocks
